### DISCUSSION TITLE
add command to power on monitors

### DISCUSSION THREAD

**Discussion by Schweber** [YaLTeR/niri#710] [Ideas]
My monitors become black after unlocking my session with `swaylock`. I would like to have a command like `niri msg action power-on-monitors` to add this as an after-resume command.

`hyprland` has got both `hyprctl dispatch dpms off` and `on` which was needed in my setup to activate my monitors after unlocking them.

**Comment by YaLTeR** [comment]
Why do they become black in the first place? Sounds like a misconfiguration.

Any input will light up the monitors.

**Reply by Schweber** [reply]
What happens is the following:

1. the monitor wakes up when i wake up my pc from sleep to unlock it (when i start typing at least)
2. after having unlocked it with swaylock, i see my workspace for a short time and then the monitor goes black again
3. i have to do another input to wake the monitor up again, thereafter it stays on

The same happened with hyprland and in its readme it is suggested to run `dpms on` as an after-resume command to swaylock (if i remember correctly, i read it somewhere for sure). This solved my problem.

I don't know how i should change the configuration of niri to prevent this behaviour. That's why i would like the option to send `dpms on` as an after-resume command so that i don't have to make a manual input to wake up the monitor again every time.

**Reply by YaLTeR** [reply]
Weird. I guess I could add an explicit power-on-monitors.

**Reply by Schweber** [reply]
That would be nice, thank you

**Reply by druskus20** [reply]
On this note. I had a similar problem. 

I connect my laptop to an external monitor at work. 

In my `kanshi` config, in my work profile, I send `niri msg output e-DP1 off` (that being the built in laptop screen) as I usually just keep my laptop lid closed. 

Then, before going home, I, without opening the lid, `systemctl suspend`. Then disconnect the extenal monitor.

When I power on the laptop, the screen is still off, but the kanshi profile for `standalone` does not apply.

## Fix: 

My solution to it, is to listen to acpi events for Lid Open and trigger `niri msg output e-DP1 on`

```
event=button/lid.*
action=runuser -l drusk -c '/usr/local/bin/lid-open-handler.sh'
```

```bash
#!/bin/bash

# logs: journalctl -u acpid.service -f
# Might need to enable: OPTIONS="-d --logevents" at /etc/default/acpid

USER=drusk
grep -q close /proc/acpi/button/lid/*/state
if [ $? = 0 ]; then
  echo 'Lid closed, doing nothing'
  exit 0
fi
grep -q open /proc/acpi/button/lid/*/state
if [ $? = 0 ]; then
  echo 'Lid opened, forcing to turn on eDP-1'
  notify-send 'Lid opened, turning on monitor'
  niri msg output eDP-1 on
fi
```

**Reply by Schweber** [reply]
thanks, i wasn't aware that i can activate an output individually. I can use this as a stopgap.

**Comment by YaLTeR** [comment]
Adding in https://github.com/YaLTeR/niri/pull/723

**Reply by Schweber** [reply]
Thank you!
